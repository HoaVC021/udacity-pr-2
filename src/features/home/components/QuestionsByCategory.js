import React from 'react';
import { Box, Grid, Typography } from '@mui/material';

import QuestionCard from './QuestionCard';

const QuestionsByCategory = ({ title, questions }) => {
  return (
    <Box sx={{ width: '100%', my: 4 }}>
      <Box sx={{ display: 'flex', justifyContent: 'center', alignItems: 'center', mb: 2 }}>
        <Typography variant='h4'>{title}</Typography>
      </Box>
      <Grid container spacing={2}>
        {questions.map((question) => (
          <Grid item key={question.id} xs={12} sm={6} md={4}>
            <QuestionCard question={question} />
          </Grid>
        ))}
      </Grid>
    </Box>
  );
};

export default QuestionsByCategory;
