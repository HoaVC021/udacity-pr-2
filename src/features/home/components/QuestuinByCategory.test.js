import React from 'react';
import { render, screen } from '@testing-library/react';
import QuestionsByCategory from './QuestionsByCategory'; // Adjust the import path as necessary

// Mock the QuestionCard component
jest.mock('./QuestionCard', () => {
  return jest.fn(() => <div>Mocked QuestionCard</div>);
});

describe('QuestionsByCategory Component', () => {
  const title = "Sample Questions";
  const questions = [
    { id: '1', author: 'Alice', timestamp: Date.now() },
    { id: '2', author: 'Bob', timestamp: Date.now() },
    { id: '3', author: 'Charlie', timestamp: Date.now() },
  ];

  it('renders title correctly', () => {
    render(<QuestionsByCategory title={title} questions={questions} />);

    expect(screen.getByText(title)).toBeInTheDocument();
  });

  it('renders correctly with no questions', () => {
    render(<QuestionsByCategory title={title} questions={[]} />);

    expect(screen.getByText(title)).toBeInTheDocument();

    const questionCards = screen.queryAllByText('Mocked QuestionCard');
    expect(questionCards.length).toBe(0);
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<QuestionsByCategory title={title} questions={questions} />);
    
    expect(asFragment()).toMatchSnapshot();
  });

  it('matches snapshot with no questions', () => {
    const { asFragment } = render(<QuestionsByCategory title={title} questions={[]} />);
    
    expect(asFragment()).toMatchSnapshot();
  });
});
