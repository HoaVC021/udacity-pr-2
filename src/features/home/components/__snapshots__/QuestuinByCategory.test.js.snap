// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionsByCategory Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-153rels"
  >
    <div
      class="MuiBox-root css-1s00k8g"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4 css-10zsxzp-MuiTypography-root"
      >
        Sample Questions
      </h4>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 css-11l5t4l-MuiGrid-root"
      />
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 css-11l5t4l-MuiGrid-root"
      />
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 css-11l5t4l-MuiGrid-root"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`QuestionsByCategory Component matches snapshot with no questions 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-153rels"
  >
    <div
      class="MuiBox-root css-1s00k8g"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4 css-10zsxzp-MuiTypography-root"
      >
        Sample Questions
      </h4>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
    />
  </div>
</DocumentFragment>
`;
